llvm/include/llvm/IR/AutoUpgrade.h

```diff
@@ -36,7 +36,8 @@ namespace llvm {
   /// for upgrading, and returns true if it requires upgrading. It may return
   /// null in NewFn if the all calls to the original intrinsic function
   /// should be transformed to non-function-call instructions.
-  bool UpgradeIntrinsicFunction(Function *F, Function *&NewFn);
+  bool UpgradeIntrinsicFunction(Function *F, Function *&NewFn,
+                                bool CanUpgradeDebugIntrinsicsToRecords = true);
```

Comment:
Nope, this part is used in this patch in `BitcodeReader.cpp:3789`, to avoid upgrading debug intrinsics to records specifically when we have the preserve flag enabled.